### PR TITLE
Deploy accountpool cr cicd

### PR DESCRIPTION
**Pull request by ArielLima**
Add AccountPool CR to the template used by CICD, having parameters per environment to keep the accountLimit
https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/7578

### CODE REVIEW

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -55,3 +57,11 @@ objects:
     account-limit: "${ACCOUNT_LIMIT}"
     root: ${ROOT_OU_ID}
     base: ${BASE_OU_ID}
+
+- apiVersion: v1
+  kind: AccountPool
+  metadata:
+    name: aws-account-operator-accountpool
+    namespace: aws-account-operator-accountpool-namespace
```
Comment (fahlmant) [inline hack/olm-registry/olm-artifacts-template.yaml:65]: I think this can be changed to `aws-account-operator` or be removed

**Comment by ArielLima** [comment]
https://issues.redhat.com/browse/OSD-4602

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 1 above)
Comment (ArielLima) [reply hack/olm-registry/olm-artifacts-template.yaml:65]: oops that was a mistake on my part, nice catch

**Comment by iamkirkbater** [comment]
So if I'm understanding this correctly, this olm-artifacts file will tell OLM what to deploy?

This structure looks correct and good to me, but I'll let someone who knows what they're really looking at review as well.

**Comment by fahlmant** [comment]
Maybe we should seperate the template out to have OLM artifacts and other (configmap, account pool, federated roles)

**Comment by ArielLima** [comment]
@fahlmant When we were doing this last time with the configmap @maorfr thought it would be best if all templates remained in one place -> https://github.com/openshift/aws-account-operator/pull/406

**Pull request merged**